openmls/src/schedule/mod.rs

```diff
@@ -361,61 +366,76 @@ impl KeySchedule {
             return Err(KeyScheduleError::InvalidState(ErrorState::NotInit));
         }
 
-        Ok(WelcomeSecret::new(
-            backend,
-            self.intermediate_secret.as_ref().unwrap(),
-        ))
+        let intermediate_secret = match &self.intermediate_secret {
```

Comment:
Can't we use `.as_ref().ok_or(KeyScheduleError::LibraryError)?` here and where we use the same pattern below? Not a strong opinion, but I somehow find these `match` patterns with `Some(x) => x` unsatisfactory.

Reply:
I guess it's a style question, but I personally find `match` statements easy to read and understand. I changed it though.